backend/src/services/course/course.service.ts

```diff
@@ -207,11 +234,35 @@ class Course {
     return course ?? null;
   }
 
-  public getMentorsByCourseId({
+  public async getMentorsByCourseId({
     courseId,
     filteringOpts,
   }: CourseGetMentorsRequestDto): Promise<UserDetailsResponseDto[]> {
-    return this.#courseRepository.getMentorsByCourseId({
+    const mentorsWithMenteesCount =
```

Comment:
What about using a query, like `.where('maxStudentCount', '>', studentCount);` or smth like that, instead of all this logic?

Reply:
@fshabanov if you mean for each calculated student count to call a query with a condition ` .where('maxStudentCount', '>', studentCount);` - I think there will be overkill, too much db calls
if you mean to create 1 query with few subqueries, it seems to be a difficult query to read, I tried to do so, but after some attempts decided to divide into separate queries and to filter programmatically in the code, don't know if it's a good approach so

Reply:
need an advice @andanfe @what1s1ove @PavelSukhinin

Reply:
I think for now we can leave current programmatic approach, because querying it from db may look prettier but more complex to handle with